realm/realm-library/src/main/java/io/realm/RealmObject.java

```diff
@@ -81,7 +82,7 @@
  */
 
 @RealmClass
-public abstract class RealmObject {
+public abstract class RealmObject<E extends RealmObject> {
```

Comment:
It kind of makes me reconsider `RealmObservable.from()` and allow users to throw anything at it, a bit like http://reactivex.io/RxJava/javadoc/rx/Observable.html#from(java.util.concurrent.Future)